mne/decoding/time_gen.py

```diff
@@ -571,16 +590,27 @@ def _init_ypred(n_train, n_test, n_orig_epochs, n_dim):
     return y_pred
 
 
-def _score_slices(y_true, list_y_pred, scorer):
+def _score_slices(y_true, list_y_pred, scorer, score_mode, cv):
     """Aux function of GeneralizationAcrossTime that loops across chunks of
     testing slices.
     """
     scores_list = list()
     for y_pred in list_y_pred:
         scores = list()
         for t, this_y_pred in enumerate(y_pred):
-            # Scores across trials
-            scores.append(scorer(y_true, np.array(this_y_pred)))
+            if score_mode in ['mean-fold-wise', 'fold-wise']:
+                # Estimate score within each fold
+                scores_ = list()
+                for train, test in cv:
+                    scores_.append(scorer(y_true[test], this_y_pred[test]))
+                # Summarize score as average across folds
+                if score_mode == 'mean-fold-wise':
+                    # Note that np.mean automatically squeezes the dimensions
+                    scores_ = np.mean(scores_, axis=0, keepdims=True)[0]
```

Comment:
Does this line get tested? I thought on old numpy keep dims didn't exist so I'm surprised Travis 2.6 is happy. Better to use np.newaxis instead for consistency.